mesonbuild/compilers/java.py

```diff
@@ -82,6 +82,12 @@ def get_buildtype_args(self, buildtype):
         return java_buildtype_args[buildtype]
 
     def compute_parameters_with_absolute_paths(self, parameter_list, build_dir):
+        for idx, i in enumerate(parameter_list):
+            if i[:4] == '-cp:' or i[:4] == '-cp;':
+                parameter_list[idx] = i[:4] + os.path.normpath(os.path.join(build_dir, i[4:]))
+            if i[:11] == '-classpath:' or i[:11] == '-classpath;':
+                parameter_list[idx] = i[:11] + os.path.normpath(os.path.join(build_dir, i[11:]))
```

Comment:
This seems to be a bit more complicated to handle. i.e. it has to handle arguments like Jar.get_classpath_args generates:
```
    def get_classpath_args(self):
        cp_paths = [os.path.join(l.get_subdir(), l.get_filename()) for l in self.link_targets]
        cp_string = os.pathsep.join(cp_paths)
        if cp_string:
            return ['-cp', os.pathsep.join(cp_paths)]
        return []
```
i.e. a list of os.pathsep separated directories in a argument following either '-cp' or '-classpath'.
like `['-cp', 'some.jar:other.jar']`

Reply:
Should be resolved now.